src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,90 @@
+import { useState } from 'react';
+
+import { filters, smartData } from './consts';
+import { FiltersMobile } from './FiltersMobile';
+
+import FiltersSvg from '@/assets/icons/filters.svg';
+import FilterRateSvg from '@/assets/icons/filter-rate.svg';
+
+import styles from './filters.module.scss';
+
+export default function Filters() {
+  const [drawerVisible, setDrawerVisible] = useState(false);
+  const [filteredProducts, setFilteredProducts] = useState<object[]>([]);
+  console.log('filteredProducts: ', filteredProducts);
+
+  const toggleDrawer = () => {
+    setDrawerVisible(!drawerVisible);
```

Comment:
`const toggleDrawer = () => setDrawerVisible(prev => !prev)
` 
Then react will use the last updatet version of drawerVisible state